pkg/api/interceptor.go

```diff
@@ -67,9 +67,23 @@ func (wa *WalletAuthorizer) Stream() grpc.StreamServerInterceptor {
 	}
 }
 
+func (wa *WalletAuthorizer) isProtocolVersion3(request *messagev1.PublishRequest) bool {
+	envelopes := request.Envelopes
+	if envelopes == nil || len(envelopes) == 0 {
+		return false
+	}
+	// If any of the envelopes are not for a v3 topic, then we treat the request as non-v3
+	for _, envelope := range envelopes {
+		if !strings.HasPrefix(envelope.ContentTopic, "/xmtp/3/") {
```

Comment:
Why if any of the envelopes are for V3. This would allow someone to get around our auth system by including a random V3 message with every request. 

I think we should change this to a requirement that _all_ envelopes must be V3

Reply:
If a request is mixed V0 and V3, and the enableV3 flag is not on, then that would disable authentication for V0 requests, which is not what we want, right? See `Test_AuthnNoTokenMixedV0V3`